frameioclient/download.py

```diff
@@ -1,18 +1,174 @@
-import requests
-import math
+import io
 import os
+import math
+import time
+import requests
+import threading
+import concurrent.futures
+
+from .utils import format_bytes
+from .exceptions import DownloadException, WatermarkIDDownloadException
+
+thread_local = threading.local()
 
 class FrameioDownloader(object):
-  def __init__(self, asset, download_folder):
+  def __init__(self, asset, download_folder, prefix, acceleration=False, concurrency=5):
+    self.acceleration = acceleration
     self.asset = asset
```

Comment:
if you don't do it already: you might want to check if the file has an `uploaded_at` which confirms that the underlying file of the asset was completely uploaded and merged... just in case someone wants to download an upload-in-progress file